test/cmd/diff.sh

```diff
@@ -110,6 +110,17 @@ run_kubectl_diff_tests() {
     # Make sure that diff --prune returns nothing (0 exit code) for 'b'.
     kubectl diff --prune -l prune-group=true -f hack/testdata/prune/b.yaml
 
+    kubectl apply --namespace nsb -l prune-group=true -f hack/testdata/prune/svc.yaml
```

Comment:
Sorry, I did not notice that there is already test for this https://github.com/kubernetes/kubernetes/blob/3ceb213fc15c347874f50027977a76ee05a1bb92/test/cmd/diff.sh#L93-L111. We are simply not testing pruning the resources not belonging to the label. That's why, we could not catch that problem via integration tests. 

Just to prevent duplication, would you mind removing tests you added recently and I'll tag the PR.

Reply:
> . We are simply not testing pruning the resources not belonging to the label. That's why, we could not catch that problem via integration tests.

The test setup might need some tweaks, but shouldn't it be possible to test? This seems like a pretty major bug for `kubectl diff`, so it would be better to have coverage for it. I'm wondering if we should also backport this fix.